org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/lookup/Scope.java

```diff
@@ -5456,6 +5464,11 @@ public void tagAsAccessingEnclosingInstanceStateOf(ReferenceBinding enclosingTyp
 						break;
 					methodDeclaration.bits &= ~ASTNode.CanBeStatic;
 				}
+				if (methodScope.referenceContext instanceof ConstructorDeclaration) {
```

Comment:
How about merging this new block into the previous one?

I believe smth like this would suffice for both cases:

```
	AbstractMethodDeclaration methodDecl = methodScope.referenceMethod();
	if (methodDecl != null && methodDecl.binding == enclosingMethod)
		break;
	if (methodDecl instanceof MethodDeclaration)
		methodDecl.bits &= ~ASTNode.CanBeStatic;
```

I arrived there after wondering, why only Constructors would need the new code. Answer: since this change methods and constructors are treated the same wrt break! :)